File: Dev/unitTest_Engine_cpp_gtest/FamilySystem/DrawnParent.cpp
#include <gtest/gtest.h>
#include <Altseed.h>
#include "../EngineTest.h"

using namespace std;
using namespace asd;

class FamilySystem_DrawnParent : public EngineTest
{
public:
	FamilySystem_DrawnParent(bool isOpenGLMode)
		: EngineTest(asd::ToAString("DrawnParent"), isOpenGLMode, 60)
	{
	}

protected:
	void OnStart()
	{
		auto other = make_shared<TextureObject2D>();
		auto parent = make_shared<TextureObject2D>();
		auto child1 = make_shared<TextureObject2D>();
		auto child2 = make_shared<TextureObject2D>();

		auto texture = asd::Engine::GetGraphics()->CreateTexture2D(asd::ToAString("Data/Texture/Cloud1.png").c_str());
		other->SetTexture(texture);
		parent->SetTexture(texture);
		child1->SetTexture(texture);
		child2->SetTexture(texture);

		other->SetColor(Color(255, 128, 128, 255));
		parent->SetColor(Color(128, 255, 128, 255));
		child1->SetColor(Color(0, 255, 255, 255));
		child2->SetColor(Color(255, 255, 0, 255));

		other->SetPosition(Vector2DF(0, 0));
		parent->SetPosition(Vector2DF(100, 0));
		child1->SetPosition(Vector2DF(0, 100));
		child2->SetPosition(Vector2DF(-100, 100));

		other->SetDrawingPriority(2);
		parent->SetDrawingPriority(3);

		parent->AddDrawnChild(child1,
			ChildManagementMode::Nothing,
			ChildTransformingMode::Position,
			ChildDrawingMode::Color);
		parent->AddDrawnChild(child2,
			ChildManagementMode::Nothing,
			ChildTransformingMode::Position,
			ChildDrawingMode::Color | ChildDrawingMode::DrawingPriority);

		Engine::AddObject2D(other);
		Engine::AddObject2D(parent);
		Engine::AddObject2D(child1);
		Engine::AddObject2D(child2);
	}

	void OnUpdating()
	{
	}

	void OnUpdated()
	{
	}

	void OnFinish()
	{
	}
};

ENGINE_TEST(FamilySystem, DrawnParent)

File: Dev/unitTest_Engine_cs/FamilySystem/DrawnParent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using asd;

namespace unitTest_Engine_cs.FamilySystem
{
    class DrawnParent : EngineTest
    {
        public DrawnParent() : base(60)
        {
        }

        protected override void OnStart()
        {
            var texture = Engine.Graphics.CreateTexture2D(CloudTexturePath);

            var other = new TextureObject2D();
            other.Texture = texture;
            other.Color = new Color(255, 128, 128, 255);
            other.Position = new Vector2DF(0, 0);
            other.DrawingPriority = 2;

            var parent = new TextureObject2D();
            parent.Texture = texture;
            parent.Color = new Color(128, 128, 128, 255);
            parent.Position = new Vector2DF(100, 0);
            parent.DrawingPriority = 3;

            var child1 = new TextureObject2D();
            child1.Texture = texture;
            child1.Color = new Color(0, 255, 255, 255);
            child1.Position = new Vector2DF(0, 100);

            var child2 = new TextureObject2D();
            child2.Texture = texture;
            child2.Color = new Color(255, 255, 0, 255);
            child2.Position = new Vector2DF(-100, 100);

            parent.AddDrawnChild(child1,
                ChildManagementMode.Nothing,
                ChildTransformingMode.Position,
                ChildDrawingMode.Color);
            parent.AddDrawnChild(child2,
                ChildManagementMode.Nothing,
                ChildTransformingMode.Position,
                ChildDrawingMode.Color | ChildDrawingMode.DrawingPriority);

            Engine.AddObject2D(other);
            Engine.AddObject2D(parent);
            Engine.AddObject2D(child1);
            Engine.AddObject2D(child2);
			Console.WriteLine("other:" + other.AbsoluteDrawingPriority);
			Console.WriteLine("parent:" + parent.AbsoluteDrawingPriority);
			Console.WriteLine("child1:" + child1.AbsoluteDrawingPriority);
			Console.WriteLine("child2:" + child2.AbsoluteDrawingPriority);
        }
    }
}
